Objects/rangeobject.c

```diff
@@ -171,6 +171,28 @@ range_dealloc(rangeobject *r)
     PyObject_Free(r);
 }
 
+static unsigned long
+get_len_of_range(long lo, long hi, long step);
+
+/// Return the length as a long, or -1 on error
+static inline long compute_range_length_long(PyObject *start, PyObject *stop, PyObject *step) {
+    int overflow = 0;
+
+    long long_start = PyLong_AsLongAndOverflow(start, &overflow);
+    if (overflow || (long_start==-1 && PyErr_Occurred()) ) {
+            return -1;
+    }
+    long long_stop = PyLong_AsLongAndOverflow(stop, &overflow);
```

Comment:
I doubt it would make a difference in practice. If we change the order, we should be consistent and also change the order of checking in `range_iter`. Let me know, I don't mind either way

Reply:
> I doubt it would make a difference in practice.

Fair enough; I think you're probably right. Let's leave as-is.